### PR TITLE
Clean up expired backfills

### PR DESCRIPTION
**Pull request by akremsa**
**What this PR does / Why we need it**:
Clean up expired backfills. This functionality should be used by synchronizer.

### CODE REVIEW

**Comment by aLekSer** [comment]
@googlebot I consent.

File: internal/statestore/backfill_test.go [hunk 1]
```diff
@@ -117,7 +120,7 @@ func TestUpdateBackfill(t *testing.T) {
 			expectedMessage: "",
 		},
 		{
-			description:     "create existing backfill, no err expected",
+			description:     "create existing backfill, no err expecteds",
```
Comment (hsorellana) [inline internal/statestore/backfill_test.go:123]: This is a typo, should be undo to what it was

Comment (akremsa) [reply internal/statestore/backfill_test.go:123]: thanks a lot

**Comment by akremsa** [comment]
@Laremere, please review

File: internal/statestore/backfill_test.go [hunk 2]
```diff
@@ -158,7 +158,7 @@ func TestUpdateBackfill(t *testing.T) {
 				require.NoError(t, err)
 			} else {
 				require.Error(t, err)
-				require.Equal(t, tc.expectedCode.String(), status.Convert(err).Code().String())
+				require.Equal(t, tc.expectedCode.String(), status.Convert(err).Code().String(), "QQQ")
```
Comment (hsorellana) [inline internal/statestore/backfill_test.go:161]: Debugging??

Comment (akremsa) [reply internal/statestore/backfill_test.go:161]: Seems yes, thanks, Hugo. 👍

File: internal/testing/e2e/backfill_test.go [hunk 3]
```diff
@@ -325,6 +325,64 @@ func TestBackfillGenerationMismatch(t *testing.T) {
 	require.Equal(t, io.EOF, err)
 }
 
+func TestCleanUpBackfills(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	om := newOM(t)
+
+	t1, err := om.Frontend().CreateTicket(ctx, &pb.CreateTicketRequest{Ticket: &pb.Ticket{}})
+	require.Nil(t, err)
```
Comment (hsorellana) [inline internal/testing/e2e/backfill_test.go:334]: probably use require.NoError??

Comment (akremsa) [reply internal/testing/e2e/backfill_test.go:334]: agree, changing it right now.

File: internal/statestore/backfill.go [hunk 4]
```diff
@@ -22,12 +22,20 @@ import (
...
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"open-match.dev/open-match/internal/ipb"
 	"open-match.dev/open-match/pkg/pb"
 )
 
+var (
+	logger = logrus.WithFields(logrus.Fields{
+		"app":       "openmatch",
+		"component": "statestore",
```
Comment (hsorellana) [inline internal/statestore/backfill.go:35]: what about using statestore.Redis?? so logs declares which implementation is using

File: internal/statestore/backfill.go [hunk 5]
```diff
@@ -143,6 +151,48 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
...
+	if err != nil {
+		return err
+	}
+
+	for _, id := range expiredBfIDs {
+		_, tickets, err := rb.GetBackfill(ctx, id)
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"error":       err.Error(),
+				"backfill id": id,
```
Comment (hsorellana) [inline internal/statestore/backfill.go:172]: probably this key should be underscored, i don't know how it outputs

File: internal/statestore/backfill.go (hunk 4 above)
Comment (akremsa) [reply internal/statestore/backfill.go:35]: ok, why not.

File: internal/statestore/backfill.go (hunk 5 above)
Comment (akremsa) [reply internal/statestore/backfill.go]: agree

**Comment by aLekSer** [comment]
Some E2E tests are failing due to update in `common.go`:
```
	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -53,3 +53,3 @@
        	            	  },
        	            	- (string) (len=21) "pendingreleasetimeout": (string) (len=6) "1000ms",
        	            	+ (string) (len=21) "pendingreleasetimeout": (string) (len=5) "200ms",
        	            	  (string) (len=26) "proposalcollectioninterval": (string) (len=5) "200ms",
```

**Review by aLekSer — APPROVED** [review]
Thanks for adding Cleanup based on Acknowledged Backfills.

File: internal/app/frontend/frontend_service.go [hunk 6]
```diff
@@ -354,7 +354,29 @@ func doWatchAssignments(ctx context.Context, id string, sender func(*pb.Assignme
 // AcknowledgeBackfill is used to notify OpenMatch about GameServer connection info.
 // This triggers an assignment process.
 func (s *frontendService) AcknowledgeBackfill(ctx context.Context, req *pb.AcknowledgeBackfillRequest) (*pb.Backfill, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	// WARNING: THIS CODE WILL BE REPLCAES BY PR "Frontend acknowledge backfill #1293"
```
Comment (aLekSer) [inline internal/app/frontend/frontend_service.go:357]:
```suggestion
	// WARNING: THIS CODE WILL BE REPLACED BY PR "Frontend acknowledge backfill #1293"
```

Comment (akremsa) [reply internal/app/frontend/frontend_service.go]: it will be replaced anyway 🤣

File: internal/statestore/backfill.go [hunk 7]
```diff
@@ -62,7 +70,7 @@ func (rb *redisBackend) CreateBackfill(ctx context.Context, backfill *pb.Backfil
 		return status.Errorf(codes.AlreadyExists, "backfill already exists, id: %s", backfill.GetId())
 	}
 
-	return acknowledgeBackfill(redisConn, backfill.GetId())
+	return nil
```
Comment (aLekSer) [inline internal/statestore/backfill.go:73]: Can you please return this back?

File: internal/app/frontend/frontend_service.go [hunk 8]
```diff
@@ -354,7 +354,62 @@ func doWatchAssignments(ctx context.Context, id string, sender func(*pb.Assignme
 // AcknowledgeBackfill is used to notify OpenMatch about GameServer connection info.
 // This triggers an assignment process.
 func (s *frontendService) AcknowledgeBackfill(ctx context.Context, req *pb.AcknowledgeBackfillRequest) (*pb.Backfill, error) {
```
Comment (sawagh) [inline internal/app/frontend/frontend_service.go:356]: I see this method also implemented in https://github.com/googleforgames/open-match/pull/1293/ - Is this accidentally in this PR too? or is this PR dependent on the other one and they need to be merged in order?

File: internal/app/synchronizer/synchronizer_service.go [hunk 9]
```diff
@@ -292,6 +292,11 @@ Registration:
 	<-closedOnCycleEnd
 	stats.Record(ctx, iterationLatency.M(float64(time.Since(cst)/time.Millisecond)))
 
+	err := s.store.CleanupBackfills(ctx)
```
Comment (sawagh) [inline internal/app/synchronizer/synchronizer_service.go:295]: for a single returned err, prefer using the more concise pattern: 

if err := s.store.CleanupBackfills(ctx); err != nil {
    logger.Errorf("Failed to clean up backfills, %s", err.Error())
}

File: internal/statestore/backfill.go [hunk 10]
```diff
@@ -143,6 +151,48 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
 	return nil
 }
 
+// CleanupBackfills removes expired backfills
+func (rb *redisBackend) CleanupBackfills(ctx context.Context) error {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
```
Comment (sawagh) [inline internal/statestore/backfill.go:157]: use similar shorter error handling pattern as suggested above (also in other places in the code)

File: internal/statestore/backfill.go [hunk 11]
```diff
@@ -143,6 +151,48 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
...
+		return status.Errorf(codes.Unavailable, "CleanupBackfills, failed to connect to redis: %v", err)
+	}
+	defer handleConnectionClose(&redisConn)
+
+	expiredBfIDs, err := rb.GetExpiredBackfillIDs(ctx)
+	if err != nil {
+		return err
```
Comment (sawagh) [inline internal/statestore/backfill.go:164]: consider adding logging here?

File: internal/statestore/backfill.go [hunk 12]
```diff
@@ -143,6 +151,48 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
...
+	}
+
+	for _, id := range expiredBfIDs {
+		_, tickets, err := rb.GetBackfill(ctx, id)
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"error":       err.Error(),
+				"backfill_id": id,
+			}).Error("cleanup failed to get the backfill")
```
Comment (sawagh) [inline internal/statestore/backfill.go:173]: does the logger already add the function name to the log? in that case, the logs could be cleaner and only state what failed ("failed to get the backfill" here)

File: internal/statestore/backfill.go [hunk 13]
```diff
@@ -143,6 +151,48 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
...
+			continue
+		}
+
+		err = rb.DeleteBackfill(ctx, id)
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"error":       err.Error(),
+				"backfill_id": id,
+			}).Error("cleanup failed to delete the backfill")
```
Comment (sawagh) [inline internal/statestore/backfill.go:182]: if you fail to delete the actual backfill (so it still has Tickets), do you still want to go ahead and delete tickets from Pending Release? would that not leave the system in inconsistent state?

File: internal/statestore/backfill_test.go [hunk 14]
```diff
@@ -604,3 +604,57 @@ func generateBackfills(ctx context.Context, t *testing.T, service Service, amoun
...
+	defer service.Close()
+	ctx := utilTesting.NewContext(t)
+
+	rc, err := redis.Dial("tcp", fmt.Sprintf("%s:%s", cfg.GetString("redis.hostname"), cfg.GetString("redis.port")))
+	require.NoError(t, err)
+
+	bfID := "mockBackfill-1"
+	ticket1ID := "t1"
+	ticket2ID := "t2"
+	ticketIDs := []string{ticket1ID, ticket2ID}
```
Comment (sawagh) [inline internal/statestore/backfill_test.go:622]: nit - since ticket1ID, ticket2ID are not used anywhere else, you might as well do []string{"t1", "t2"}

File: internal/statestore/backfill_test.go [hunk 15]
```diff
@@ -604,3 +604,57 @@ func generateBackfills(ctx context.Context, t *testing.T, service Service, amoun
...
+	bfLastAck := "backfill_last_ack_time"
+	proposedTicketIDs := "proposed_ticket_ids"
+
+	// ARRANGE
```
Comment (sawagh) [inline internal/statestore/backfill_test.go:626]: nit: inconsistent comments through the test. Prefer using complete sentences - you may just choose to describe a comment blurb for the test rather than commenting each method as the names are verbose anyways.

File: internal/statestore/backfill_test.go [hunk 16]
```diff
@@ -604,3 +604,57 @@ func generateBackfills(ctx context.Context, t *testing.T, service Service, amoun
...
+	require.NoError(t, err)
+
+	err = service.AddTicketsToPendingRelease(ctx, ticketIDs)
+	require.NoError(t, err)
+
+	// ACT
+	err = service.CleanupBackfills(ctx)
+	require.NoError(t, err)
+
+	// ASSERT
```
Comment (sawagh) [inline internal/statestore/backfill_test.go:644]: missing any actual assert here?

File: internal/statestore/backfill_test.go [hunk 17]
```diff
@@ -604,3 +604,57 @@ func generateBackfills(ctx context.Context, t *testing.T, service Service, amoun
...
+
+	// backfill doesn't exist anymore
+	_, _, err = service.GetBackfill(ctx, bfID)
+	require.Error(t, err)
+	require.Equal(t, "Backfill id: mockBackfill-1 not found", status.Convert(err).Message())
```
Comment (sawagh) [inline internal/statestore/backfill_test.go:649]: simply checking error message equality is sufficient? it would fail anyways if there is no error.

File: internal/testing/e2e/backfill_test.go [hunk 18]
```diff
@@ -177,31 +178,23 @@ func TestProposedBackfillCreate(t *testing.T) {
...
 	actual, err := om.Frontend().GetBackfill(ctx, &pb.GetBackfillRequest{BackfillId: bfID})
-	require.Nil(t, err)
+	require.NoError(t, err)
 	require.NotNil(t, actual)
 
 	b.Id = actual.Id
 	b.CreateTime = actual.CreateTime
 	require.True(t, proto.Equal(b, actual))
-
-	client, err := om.Query().QueryTickets(ctx, &pb.QueryTicketsRequest{Pool: &pb.Pool{
```
Comment (sawagh) [inline internal/testing/e2e/backfill_test.go:198]: curious why this blurb was removed? seems unrelated to this change itself?

File: internal/testing/e2e/backfill_test.go [hunk 19]
```diff
@@ -318,13 +303,69 @@ func TestBackfillGenerationMismatch(t *testing.T) {
...
+		Config:  om.MMFConfigGRPC(),
+		Profile: &pb.MatchProfile{},
+	})
+	require.NoError(t, err)
+
+	_, err = stream.Recv()
+	require.Error(t, err)
+	e, ok := status.FromError(err)
+	require.True(t, ok)
+	require.Contains(t, e.Message(), "error(s) in FetchMatches call. syncErr=[failed to handle match backfill: 1: rpc error: code = NotFound desc = Backfill id:")
```
Comment (sawagh) [inline internal/testing/e2e/backfill_test.go:366]: Curious about the test logic here. Given your sleep is even before the FetchMatches call - how is that sleep causing backfill expiration? I would expect FetchMatches call to trigger MMF and have a backfill generated and so shouldnt the sleep be after the call?

File: internal/testing/e2e/common.go [hunk 20]
```diff
@@ -162,15 +162,15 @@ func (om *om) MMFConfigHTTP() *pb.FunctionConfig {
 // that parsing bugs can't hide logic bugs.
 const registrationInterval = time.Millisecond * 200
 const proposalCollectionInterval = time.Millisecond * 200
-const pendingReleaseTimeout = time.Millisecond * 200
+const pendingReleaseTimeout = time.Second * 1
```
Comment (sawagh) [inline internal/testing/e2e/common.go:165]: why change these from 200ms to 1s?

**Review by sawagh — CHANGES_REQUESTED** [review]

File: internal/app/frontend/frontend_service.go (hunk 8 above)
Comment (akremsa) [reply internal/app/frontend/frontend_service.go:356]: yes, the whole frontend_service.go file was taken from #1293

File: internal/testing/e2e/common.go (hunk 20 above)
Comment (akremsa) [reply internal/testing/e2e/common.go]: This was introduced in #1310 and I needed this change too.

File: internal/statestore/backfill.go (hunk 10 above)
Comment (akremsa) [reply internal/statestore/backfill.go:214]: I suggest leaving this one as is, and in other places in statestore too. If I change it to the shorter pattern you suggest then redisConn won't be visible out of this if-scope. Then I need to create a separate variable for that. So I don't think it's worth changing, we follow this pattern everywhere here. This is a matter of personal preference.

Comment (sawagh) [reply internal/statestore/backfill.go:214]: Sorry. For methods that return value along with error (like here), it is reasonable to have the error check outside (like you have already). 

For methods that return just err, I think the shorter format is concise and readable. However, a quick check tells me we are using either format in the code - not uniform but I am ok with you not changing the format.

File: internal/statestore/backfill.go [hunk 21]
```diff
@@ -200,6 +208,48 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
 	return nil
 }
 
+// CleanupBackfills removes expired backfills
+func (rb *redisBackend) CleanupBackfills(ctx context.Context) error {
```
Comment (scosgrave) [inline internal/statestore/backfill.go:212]: Should this function mirror the code in frontend_service.go's doDeleteBackfill? If so, I think you are missing the DeindexBackfill call.  We might also want to find a way to write the code once and re-use it.

File: internal/testing/e2e/backfill_test.go (hunk 19 above)
Comment (scosgrave) [reply internal/testing/e2e/backfill_test.go:366]: I'm pretty sure the wait prior to FetchMatches is valid. The Backfill is created right after the call to CreateTicket at the start of the test. The Backfill is acknowledged right when it is created, and that sets its expiration timestamp, so waiting prior to calling FetchMatches can cause the Backfill to age.
I think this test mirrors the case where a game server has a player leave a game, and it creates the Backfill on its own. Maybe we need another test where the MMF is not able to completely fill a Match, and requests a Backfill to be created?

File: internal/statestore/backfill.go (hunk 21 above)
Comment (scosgrave) [inline internal/statestore/backfill.go:212]: Sorry for asking this here, and not in backfill.go, but I think it is related closely enough:

Why do we expire Backfills at 80% of the pendingReleaseTimeout? I think it isn't very clear that setting pendingReleaseTimeout will have an effect on when Backfills are deleted. I think it would be better for Backfills to have their own explicit expiration timeout in the config, and to update the ticket release process to disassociate the ticket from a Backfill, if it was associated with one.

File: internal/testing/e2e/backfill_test.go (hunk 19 above)
Comment (sawagh) [reply internal/testing/e2e/backfill_test.go:366]: Thanks. That case makes sense - yes I was confusing it with the MMF created backfill case - and +1 to having a test case to cover that.

File: internal/statestore/backfill.go (hunk 11 above)
Comment (akremsa) [reply internal/statestore/backfill.go:221]: In case of **returning** an error it is better to log one outside of statestore package.

File: internal/statestore/backfill.go (hunk 12 above)
Comment (akremsa) [reply internal/statestore/backfill.go:230]: It does not.
Example of the current output:
`cleanup failed to get the backfill            app=openmatch backfill_id=mockBackfill-1 component=statestore.redis error=QQQ
`

File: internal/statestore/backfill_test.go (hunk 14 above)
Comment (akremsa) [reply internal/statestore/backfill_test.go:622]: ok

File: internal/statestore/backfill_test.go (hunk 15 above)
Comment (akremsa) [reply internal/statestore/backfill_test.go:626]: "Arrange act assert" is a common pattern of writing tests and I've just made the test a bit more readable. You clearly see that `err = service.AddTicketsToPendingRelease(ctx, ticketIDs)`, for example, is a part of an ARRANGE section, which means it is used to prepare data for an actual test. Then I clearly separate a function under the test with ACT section, and so on. So I find these comments quite useful here.

File: internal/statestore/backfill_test.go (hunk 16 above)
Comment (akremsa) [reply internal/statestore/backfill_test.go:644]: see comment above

File: internal/statestore/backfill_test.go (hunk 17 above)
Comment (akremsa) [reply internal/statestore/backfill_test.go:649]: Checking for an error explicitly provides more precise message.
Example:
`Error:      	An error is expected but got nil.`
I clearly see that there should be an error but the function returned nil.

If I leave only comparing error messages I will get:
```        	
Error:      	Not equal: 
expected: "Backfill id: mockBackfill-1 not found"
actual  : ""
```
And here I'm not sure - is error nil or is its description just empty for some reason?

File: internal/testing/e2e/backfill_test.go (hunk 18 above)
Comment (akremsa) [reply internal/testing/e2e/backfill_test.go:198]: Had issues with this test, agreed with @yeukovichd to remove these lines at some point, but this test is updated in #1318 so I will return this code back.

File: internal/statestore/backfill.go (hunk 21 above)
Comment (akremsa) [reply internal/statestore/backfill.go:212]: We had a conversation with @Laremere in one of the PRs thread and agreed on using this 80%. 
@aLekSer we do plan to add this value to a separate config parameter, don't we?

Comment (akremsa) [reply internal/statestore/backfill.go:212]: good catch. I will try to reuse that code.

Comment (scosgrave) [reply internal/statestore/backfill.go:212]: Ok. Thanks for discussing it with @Laremere. If this was the agreed-upon approach, then I'm fine with it.

File: internal/statestore/backfill.go (hunk 13 above)
Comment (akremsa) [reply internal/statestore/backfill.go:239]: we try to delete as much as we can

**Comment by akremsa** [comment]
@scosgrave @sawagh I've updated this PR, could you please check it again?

File: internal/statestore/backfill.go [hunk 22]
```diff
@@ -200,6 +208,77 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
...
+
+	// 1. deindex backfill
+	err = rb.DeindexBackfill(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	// just log errors and try to perform as mush actions as possible
+
+	go func() {
```
Comment (yeukovichd) [inline internal/statestore/backfill.go:227]: Why is new goroutine required here?

Comment (akremsa) [reply internal/statestore/backfill.go]: removed

File: internal/statestore/backfill_test.go (hunk 15 above)
Comment (sawagh) [reply internal/statestore/backfill_test.go:629]: I like the test pattern itself - but did not think the comments were necessary. Comments are only useful when they add value not obvious when reading the code. This pattern seems clean enough and the comments // ARRANGE // ACT // ASSERT do not seem to add value. If anything, they do seem misplaced compared to any other comment in production code. 

Did find an interesting blog that goes through this exact pattern and why you might add comments - and why it is a bad idea. Interesting read: https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80

Readability is subjective. You may leave them in place if you think it makes it more readable. It did not make it any more readable for me.

File: internal/statestore/backfill_test.go (hunk 17 above)
Comment (sawagh) [reply internal/statestore/backfill_test.go:662]: In your sample output, the the actual error being "" is clearly indicative of the error returned being empty right? Still not sure what extra value the extra check adds. The test expects an error - and the returned error is not that - and thats the failure. Unless this error does not catch some output, I do not see the value for the extra check.

File: internal/testing/e2e/backfill_test.go (hunk 18 above)
Comment (sawagh) [reply internal/testing/e2e/backfill_test.go]: Will this be returned back in this same PR? Please resolve the comment when it is reverted back. Thanks!

**Review by sawagh — APPROVED** [review]
I still have some pending comments - but comments are not blocking. Feel free to check in once Scott signs off as well.

File: internal/statestore/backfill_test.go (hunk 17 above)
Comment (akremsa) [reply internal/statestore/backfill_test.go:662]: Message `actual  : ""` doesn't give an information where it was nil error returned or actual error with an empty string description. `status.Convert(err).Message()` converts nil error to an empty string.

File: internal/testing/e2e/backfill_test.go (hunk 18 above)
Comment (akremsa) [reply internal/testing/e2e/backfill_test.go]: This test is already updated in #1318

File: internal/statestore/backfill_test.go (hunk 17 above)
Comment (sawagh) [reply internal/statestore/backfill_test.go:662]: Does it matter if it returned nil - or "" - both of which are unexpected when the expected value is "Backfill id: mockBackfill-1 not found"? What debugging value is added by understanding that it returned nil - over not being sure if it was nil or empty string? I wont block subject my approval over this - but am not convinced that the added check adds any value to the test.

**Comment by hsorellana** [comment]
LGTM 👍

File: internal/statestore/backfill.go [hunk 23]
```diff
@@ -200,6 +208,77 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
 	return nil
 }
 
+// DeleteBackfillCompletely using goroutine performs a set of operations to remove backfill and all related entities.
```
Comment (aLekSer) [inline internal/statestore/backfill.go:211]: Does it actually uses go routines?

File: internal/testing/e2e/backfill_test.go [hunk 24]
```diff
@@ -533,6 +534,65 @@ func TestBackfillGenerationMismatch(t *testing.T) {
...
+		require.False(t, ok)
+
+		out <- m.MatchId
+		return nil
+	})
+
+	// wait until backfill is expired, then try to get it
+	time.Sleep(pendingReleaseTimeout * 2)
+
+	// statestore.CleanupBackfills is called at the end of each syncronizer cycle after fetch matches call, so expired backfill will be removed
```
Comment (aLekSer) [inline internal/testing/e2e/backfill_test.go:579]:
```suggestion
	// statestore.CleanupBackfills is called at the beginning of each syncronizer cycle after fetch matches call, so expired backfill will be removed
```

**Comment by aLekSer** [comment]
I have created #1324 ticket to track some contention issues which would arise if we move this logic into go routine. Mostly FetchMatches can return the Backfill to the Director, but there would be no Backfills already if cleanup occurred before this event. We would investigate this deeper in Scale tests.

File: internal/testing/e2e/backfill_test.go [hunk 25]
```diff
@@ -533,6 +534,65 @@ func TestBackfillGenerationMismatch(t *testing.T) {
...
+	stream, err := om.Backend().FetchMatches(ctx, &pb.FetchMatchesRequest{
+		Config:  om.MMFConfigGRPC(),
+		Profile: &pb.MatchProfile{},
+	})
+	require.NoError(t, err)
+
+	_, err = stream.Recv()
+	e, ok := status.FromError(err)
+	require.True(t, ok)
+	require.Contains(t, e.Message(), "error(s) in FetchMatches call. syncErr=[failed to handle match backfill: 1: rpc error: code = NotFound desc = Backfill id:")
```
Comment (aLekSer) [inline internal/testing/e2e/backfill_test.go:589]: @akremsa this should not return an error.

Comment (aLekSer) [reply internal/testing/e2e/backfill_test.go:589]: We can make 2 FetchMatches requests.
